plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy

```diff
@@ -375,12 +375,18 @@ class AzBatchService implements Closeable {
                 .trim()
                 .replaceAll(/[^a-zA-Z0-9-_]+/,'_')
 
-        final String key = "job-${Rnd.hex()}-${name}"
+        final String key = includeTowerPrefix("job-${Rnd.hex()}-${name}", System.getenv())
         // Azure batch job max len is 64 characters, however we keep it a bit shorter
         // because the jobId + taskId composition must be less then 100
         final MAX_LEN = 62i
         return key.size()>MAX_LEN ? key.substring(0,MAX_LEN) : key
     }
+    protected static String includeTowerPrefix(String name, Map<String,String> environment) {
```

Comment:
Maybe, this can be turned into a common utility method used by all implementations. Something like 

```
class TaskUtils {
  
   static String includeTowerPrefix(String name, Map<String,String> env) { .. } 
 
}
```

Reply:
Class TaskUtils created